Increase test coverage

S2dentik opened the pull request.

Commit:
Add test cases

Add test for unexisting help file
Add test for no 'path' key inside caprice dictionary
Add test for checking whether default exclude file path is empty when no 'path' key
Add test for running Pacman with empty array of paths which should return early and not crash
Add test for Scissors returning empty tree when given unexistent path
Add test for deserializer returning nil when given dictionary doesn't have necessary
Add test for running JSON serializer with a wrong input/encoding

Commit:
Allow filename to be empty when reporter is Xcode

thelvis4 commented on TaylorFrameworkTests/CapriceTests/Tests/FlagsTests.swift:
```diff
@@ -10,6 +10,15 @@ import Quick
 @testable import TaylorFramework
 
+class HelpFlagMock: HelpFlag {
+    func executeTest() -> Bool {
```
Could we move this method in a helper class in Tests bundle?  
It should be not to modify the production code just because we need to test it were possible.

Commit:
Remove HelpFlagMock class

Remove HelpFlagMock class as testing must not affect production code
intentionally.
